### PR TITLE
Refactor path creation on DefaultLinkWidget.tsx

### PR DESCRIPTION
**Pull request by DrummerHead**
On my quest to see if I can get to have optional ending arrows on links, I ended up refactoring path generation so that I can better understand it :)

Functionality is the same; commas added on M, L and C paths in accordance to https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d (although it makes no difference on rendered output, it nicely ties together pairs of numbers to better grok that they're coordinates)

Cheers!

### CODE REVIEW

**Comment by dylanvorster** [comment]
will have a look when I'm back from holiday o/

**Comment by DrummerHead** [comment]
Great! Enjoy your time :D

File: src/defaults/DefaultLinkWidget.tsx [hunk 1]
```diff
@@ -16,6 +16,23 @@ export interface DefaultLinkState {
 	selected: boolean;
 }
 
+interface Point {
```
[the remark this answers is not in the document]
Comment (DrummerHead) [reply src/defaults/DefaultLinkWidget.tsx]: Done! https://github.com/projectstorm/react-diagrams/pull/92/commits/c8116e12ca16ca57208b0c00a2fc9fadf30f35fd

File: src/defaults/DefaultLinkWidget.tsx [hunk 2]
```diff
@@ -16,6 +16,23 @@ export interface DefaultLinkState {
...
+	x: number;
+	y: number;
+}
+
+const linePath = (firstPoint: Point, lastPoint: Point): string =>
+	`M${firstPoint.x},${firstPoint.y} L ${lastPoint.x},${lastPoint.y}`;
+
+const curvePath = (
```
[the remark this answers is not in the document]
Comment (DrummerHead) [reply src/defaults/DefaultLinkWidget.tsx]: These being pure functions that just take parameters and return; and they don't need to access `this`, it's generally recommended to not make them class methods.

I could make them static methods, but then it gets more complicated to access them inside their own class.

In https://github.com/projectstorm/react-diagrams/pull/92/commits/ad7ccf07e5ff40b8390af260ed7b27cea59cae4e I moved the functions inside the class in the spirit of keeping the project style consistent :) and the functionality is kept the same.

**Comment by DrummerHead** [comment]
I think that amazingly, I just did the changes you requested 11 minutes after you merged the original pull request. Crazy!

**Pull request merged**